README.md

```diff
@@ -1,76 +1,129 @@
 ![](./resources/official_armmbed_example_badge.png)
-## Getting started with the network-socket API
+# Socket Example
 
-(Note: To see this example in a rendered form you can import into the Arm Mbed Online Compiler, please see [the documentation](https://os.mbed.com/docs/mbed-os/latest/apis/socket.html#socket-example).)
+This example shows usage of [network-socket API](https://os.mbed.com/docs/latest/reference/network-socket.html).
 
-This is a quick example of a simple HTTP client program using the
-[network-socket API](https://os.mbed.com/docs/latest/reference/network-socket.html) that [Mbed OS](https://github.com/ARMmbed/mbed-os) provides.
+The program brings up an underlying network interface and if it's Wifi also scans for access points.
+It creates a TCPSocket and performs an HTTP transaction targeting the website in the `mbed_app.json` config.
 
-The program brings up an underlying network interface, and uses it to perform an HTTP
-transaction over a TCPSocket.
+(Note: To see this example in a rendered form you can import into the Arm Mbed Online Compiler,
+please see [the documentation](https://os.mbed.com/docs/mbed-os/latest/apis/socket.html#socket-example).)
 
 ### Selecting the network interface
 
-This application is able to use any network inteface it finds. Please see the Mbed OS documentationg for [selecting the default network interface](https://os.mbed.com/docs/v5.10/apis/network-interfaces.html).
+This application is able to use any network interface it finds.
+Please see the Mbed OS documentation for [selecting the default network interface](https://os.mbed.com/docs/v5.10/apis/network-interfaces.html).
 
-For example, building on Ethernet enabled boards, you do not do any configuration.
+For example, building on Ethernet enabled boards, you do not need any further configuration.
 
-Building for WiFi boards, you need to provide SSID, password and security settings in `mbed_app.json` as instructed in the documentation. For example, like this:
+### WiFi
+
+If you want to use WiFi you need to provide SSID, password and security settings in `mbed_app.json`.
+
+If your board doesn't provide WiFi as the default interface because it has multiple interfaces you need to specify that you want WiFi in `mbed_app.json`.
 
 ```
 {
     "target_overrides": {
         "*": {
-            "platform.stdio-convert-newlines": true,
             "target.network-default-interface-type": "WIFI",
```

Comment:
showing the rest is no longer needed as the default config includes those entries now